src/flaresolverr_service.py

```diff
@@ -287,20 +287,37 @@ def click_verify(driver: WebDriver):
     time.sleep(2)
 
 
+def get_correct_window(driver: WebDriver) -> WebDriver:
+    if len(driver.window_handles) > 1:
+        for window_handle in driver.window_handles:
+            driver.switch_to.window(window_handle)
+            current_url = driver.current_url
+            if not current_url.startswith("devtools://devtools"):
+                return driver
+    return driver
+
+
+def access_page(driver: WebDriver, url: str) -> None:
+    driver.get(url)
+    driver.start_session()
+
+
 def _evil_logic(req: V1RequestBase, driver: WebDriver, method: str) -> ChallengeResolutionT:
     res = ChallengeResolutionT({})
     res.status = STATUS_OK
     res.message = ""
 
+
     # navigate to the page
     logging.debug(f'Navigating to... {req.url}')
-    driver.get(req.url)
-    driver.start_session() # required to bypass Cloudflare
     if method == 'POST':
+        access_page(driver, req.url)
         _post_request(req, driver)
     else:
-        driver.get(req.url)
-        driver.start_session()  # required to bypass Cloudflare
+        access_page(driver, req.url)
+        driver.start_session() # required to bypass Cloudflare
```

Comment:
Why to call start_session() again?

Reply:
https://github.com/FlareSolverr/FlareSolverr/pull/1065#discussion_r1488333820